src/opera_disp_tms/prep_stack.py

```diff
@@ -0,0 +1,146 @@
+    """Load the short wavelength displacement data for and OPERA DISP granule.
+    Clips to frame map and masks out invalid data.
+
+    Args:
+        granule: The granule to load
+        bbox: The bounding box to clip to
+
+    Returns:
+        The short wavelength displacement data as an xarray DataArray
+    """
+    datasets = ['short_wavelength_displacement', 'recommended_mask']
+    with s3_xarray_dataset(granule.s3_uri) as ds:
+        granule_xr = open_opera_disp_granule(ds, granule.s3_uri, datasets)
+        granule_xr = granule_xr.rio.clip_box(*bbox, crs='EPSG:3857')
+        granule_xr = granule_xr.load()
+        valid_data_mask = granule_xr['recommended_mask'] == 1
+        sw_cumul_disp_xr = granule_xr['short_wavelength_displacement'].where(valid_data_mask, np.nan)
+        sw_cumul_disp_xr.attrs = granule_xr.attrs
+    sw_cumul_disp_xr.attrs['bbox'] = bbox
+    return sw_cumul_disp_xr
+
+
+def check_connected_network(granule_xrs: list[xr.DataArray]) -> None:
+    """Check that cumulative displacement can reconstructed using given granule set.
+
+    Args:
+        granules_xr: A list of granule xarray DataArrays
+    """
+    reference_dates = sorted(list({g.reference_date for g in granule_xrs}))
+    secondary_dates = sorted(list({g.secondary_date for g in granule_xrs}))
+    for reference_date in reference_dates[1:]:
+        if not any([within_one_day(reference_date, s) for s in secondary_dates]):
+            raise ValueError('Granule network is unconnected')
+
+
+def align_to_common_reference_date(granule_xrs: list[xr.DataArray], start_date: datetime) -> None:
+    """Align granules to a common reference date, correcting reference date changes.
+
+    Args:
+        granule_xrs: A list of granule xarray DataArrays
+        start_date: The start date to align to
+    """
+    granule_xrs.sort(key=lambda x: x.secondary_date)
+    check_connected_network(granule_xrs)
+
+    if start_date <= granule_xrs[0].reference_date:
+        zero_xr = granule_xrs[0].copy()
+        zero_xr.attrs['secondary_date'] = zero_xr.attrs['reference_date']
+        zero_xr.data = np.zeros(zero_xr.shape)
+        granule_xrs.insert(0, zero_xr)
+
+    previous_granule_xr = granule_xrs[0]
+    correction = -previous_granule_xr.data
+
+    for granule_xr in granule_xrs:
+        if not within_one_day(granule_xr.reference_date, previous_granule_xr.reference_date):
+            correction = previous_granule_xr.data
+        granule_xr += correction
+        previous_granule_xr = granule_xr
+
+    for granule_xr in granule_xrs:
+        granule_xr.attrs['reference_date'] = granule_xrs[0].secondary_date
+
+
+def load_sw_disp_stack(
+    frame_id: int, bbox: tuple[float, float, float, float], begin_date: datetime, end_date: datetime, strategy: str
+):
+    """Load the short wavelength displacement data for a frame and date range.
+    Update the reference date of all granules to the earliest reference date.
```

Comment:
"update the reference date" still isn't how I like to describe the operation, although it's not a blocker for me. Maybe describe it in terms of what the returned data will look like, rather than what action we took to prepare the data?

> Load a short wavelength displacement time series for a given frame and date range, with all data values relative to the earliest available reference date